src/uu/pgrep/src/pgrep.rs

```diff
@@ -0,0 +1,126 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+pub mod pid;
+
+use clap::{arg, crate_version, Arg, ArgAction, Command};
+use pid::walk_pid;
+use uucore::{error::UResult, format_usage, help_about, help_usage};
+
+const ABOUT: &str = help_about!("pgrep.md");
+const USAGE: &str = help_usage!("pgrep.md");
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+
+    let default_delimiter = "\n".into();
+    let delimiter = matches
+        .get_one::<String>("delimiter")
+        .unwrap_or(&default_delimiter);
+
+    let mut programs = matches
+        .get_many::<String>("program-name")
+        .expect("need at least one program-name")
+        .map(|it| it.to_string())
+        .collect::<Vec<_>>();
```

Comment:
My suggestion is to use `pattern` instead of `programs` and `program-name` because it's the name used in the man page.